### PR TITLE
add aks config - chart config

### PR DESCRIPTION
**Pull request by mdayican**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/BPS-838

### Change description ###

add aks config - chart config

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: charts/rpe-send-letter-service/values.yaml [hunk 1]
```diff
@@ -1,37 +1,58 @@
...
+
     # db - migration
-    FLYWAY_URL: "jdbc:postgresql://{{ .Release.Name }}-postgresql:5432/{{ .Values.postgresql.postgresqlDatabase}}"
-    FLYWAY_NOOP_STRATEGY: "false"
+    FLYWAY_URL: "jdbc:postgresql://rpe-send-letter-service-db-{{ .Values.global.environment }}.postgres.database.azure.com:5432/send_letter?sslmode=require"
+
     # encryption
-    ENCRYPTION_ENABLED: "false"
-    ENCRYPTION_PUBLIC_KEY: ""
+    ENCRYPTION_ENABLED: "true"
```
Comment (rkondratowicz) [inline charts/rpe-send-letter-service/values.yaml]: wont work without a key

Comment (mdayican) [reply charts/rpe-send-letter-service/values.yaml]: removed. it is in bootstrap.

File: Jenkinsfile_CNP [hunk 2]
```diff
@@ -33,19 +33,17 @@ static LinkedHashMap<String, Object> secret(String secretName, String envVar) {
...
+
+  // vars needed for AKS testing
+  env.TEST_S2S_NAME = 'send_letter_tests'
+  env.TEST_S2S_URL = 'http://rpe-service-auth-provider-aat.service.core-compute-aat.internal'
+  env.TEST_FTP_HOSTNAME = 'cmseft.services.xerox.com'
+  env.TEST_FTP_PORT = 22
+  env.TEST_FTP_FINGERPRINT = 'SHA256:3tX3DIkqd1Loz2alHfnt+qjHocfxk0YUOZHlnf9Zgdk'
+  env.TEST_FTP_REPORTS_FOLDER = 'FROM_XEROX'
+  env.TEST_ENCRYPTION_ENABLED = false
+  env.TEST_FTP_TARGET_FOLDER = 'TO_XEROX'
```
Comment (rkondratowicz) [inline Jenkinsfile_CNP]: what happened here? this one didn't exist before ( `FTP_TARGET_FOLDER` did )

File: charts/rpe-send-letter-service/values.yaml (hunk 1 above)
Comment (rkondratowicz) [reply charts/rpe-send-letter-service/values.yaml]: why is it enabled now?

File: charts/rpe-send-letter-service/values.yaml [hunk 3]
```diff
@@ -1,37 +1,58 @@
...
     FTP_DOWNTIME_FROM: "23:58"
     FTP_DOWNTIME_TO: "23:59"
-    # mails
-    SPRING_MAIL_HOST: "false"
+
+    #smtp - For prod turn it on and assign values to vault reports-email-username and password
+    SMTP_HOST:  "false"
+
+    #reports. depends on smtp - For prod overwrite
+    UPLOAD_SUMMARY_REPORT_CRON: "0 0 19 * * *"
```
Comment (rkondratowicz) [inline charts/rpe-send-letter-service/values.yaml]: is this needed?

File: Jenkinsfile_CNP (hunk 2 above)
Comment (mdayican) [reply Jenkinsfile_CNP]: add TEST prefix to clarify and it interferes with integration test case when removing before('smoketest:preview')

File: charts/rpe-send-letter-service/values.preview.template.yaml [hunk 4]
```diff
@@ -2,5 +2,27 @@ java:
...
+    # db - migration
+    FLYWAY_URL: "jdbc:postgresql://{{ .Release.Name }}-postgresql:5432/{{ .Values.postgresql.postgresqlDatabase}}"
+    FLYWAY_NOOP_STRATEGY: "false"
+    # encryption
+    ENCRYPTION_ENABLED: "false"
 postgresql:
   enabled: true
+
+
+
```
Comment (rkondratowicz) [inline charts/rpe-send-letter-service/values.preview.template.yaml]: can be removed

File: charts/rpe-send-letter-service/values.yaml [hunk 5]
```diff
@@ -1,37 +1,58 @@
...
     # schedules
     SCHEDULING_ENABLED: "true"
+    SCHEDULING_LOCK_AT_MOST_FOR: "PT10M"
+
     # inherited
-    S2S_NAME: "send_letter_tests"
-    S2S_URL: "http://rpe-service-auth-provider-aat.service.core-compute-aat.internal"
+    S2S_URL: "http://rpe-service-auth-provider-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal"
+
+    FILE_CLEANUP_ENABLED: "true"
```
Comment (rkondratowicz) [inline charts/rpe-send-letter-service/values.yaml]: will this be enabled on prod?

File: charts/rpe-send-letter-service/values.yaml (hunk 1 above)
Comment (doncem) [reply charts/rpe-send-letter-service/values.yaml]: encryption is and should remain "on". for previews it's ok to have disabled. such situation **is** represented in this pr

File: charts/rpe-send-letter-service/values.yaml (hunk 5 above)
Comment (doncem) [reply charts/rpe-send-letter-service/values.yaml]: ideally we **only** want this on aat. so would be good to have `false` here and override for aat (if possible?)

File: charts/rpe-send-letter-service/values.yaml (hunk 3 above)
Comment (doncem) [reply charts/rpe-send-letter-service/values.yaml]: we can probably ignore it (remove from here). same with `FILE_CLEANUP_CRON`

File: charts/rpe-send-letter-service/values.yaml [hunk 6]
```diff
@@ -1,37 +1,58 @@
...
   environment:
     # db
-    LETTER_TRACKING_DB_HOST: "{{ .Release.Name }}-postgresql"
-    LETTER_TRACKING_DB_NAME: "{{ .Values.postgresql.postgresqlDatabase}}"
-    LETTER_TRACKING_DB_USER_NAME: "{{ .Values.postgresql.postgresqlUsername}}"
-    LETTER_TRACKING_DB_PASSWORD: "{{ .Values.postgresql.postgresqlPassword}}"
+    LETTER_TRACKING_DB_HOST: "rpe-send-letter-service-db-{{ .Values.global.environment }}.postgres.database.azure.com"
+    LETTER_TRACKING_DB_PORT: "5432"
+    LETTER_TRACKING_DB_NAME: "send_letter"
+    LETTER_TRACKING_DB_USER_NAME: "send_letter@rpe-send-letter-service-db-{{ .Values.global.environment }}"
```
Comment (doncem) [inline charts/rpe-send-letter-service/values.yaml]: I still don't like hardcoded values here. Would it make sense if we load from vault and move to bootstrap as well?

**Comment by mdayican** [comment]
> ingres host config is missing from `values.yaml`. Refer bulk scan processor/orchestrator

add ingressHost. used "rpe-send-letter-service"

**Comment by mdayican** [comment]
> Let's refer to the PR where `FILE_CLEANUP_ENABLED` is enabled for aat

https://github.com/hmcts/cnp-flux-config/pull/1017

**Pull request merged**